slam3d/core/Graph.hpp

```diff
@@ -395,7 +426,7 @@ namespace slam3d
 		 * It should not be used directly, but is used internally.
 		 * @param v VertexObject to be stored in the graph
 		 */
-		virtual void addVertex(const VertexObject& v) = 0;
+		virtual void addVertex(const VertexObject& v, Measurement::Ptr measurement) = 0;
```

Comment:
> I think giving the VertexObject here is not really necessary anymore, isn't it? It should suffice to give the Measurement::Ptr and create the VertexObject internally.

True, it was even unused in the cpp -> removed




> I wonder if all these changes to slam3d::Graph are actually required. It shouldn't affect the Graph if the VertexObject's hold an UUID instead of a Pointer, or do I miss something here?

True, now there is only the new contructor of the VertexObject anf the getAllVertices() function left. The new  VertexObject contructor (Measurement::Ptr) forces to copy the metadata from the measurement to the Vertexobject and created an compiler error in case old code tries to add an VertexObject without Measurement uuid.

Reply:
Would have been nice, but BGL demands that VertexObject has a default constructor.